core/templates/pages/contributor-dashboard-page/modal-templates/translation-suggestion-review-modal.component.ts

```diff
@@ -301,6 +391,21 @@ export class TranslationSuggestionReviewModalComponent implements OnInit {
     this.init();
   }
 
+  showNextItemToReview(suggestionId: string): void {
+    this.resolvedSuggestionIds.push(this.activeSuggestionId);
+
+    // Remove resolved contributions from the record.
+    delete this.allContributions[this.activeSuggestionId];
+
+    // If the reviewed item is the last item, close the modal.
+    if (this.lastSuggestionToReview || this.isLastItem) {
+      this.activeModal.close(this.resolvedSuggestionIds);
+      return;
+    }
```

Comment:
Should this logic be moved to `gotoNextItem`?

Reply:
I think this logic should stay in this function as it processes the 'review' logic. The `gotoNextItem` function is not meant to navigate (or do anything) if we are on the last item, however reviewing the last (or the only) item should close the modal.

Reply:
Yeah, okay, makes sense.